_posts/2020-06-11-Solidity-0610-release-announcement.md

```diff
@@ -0,0 +1,126 @@
+
+Please upgrade if you are using internal library functions with calldata paramaters in connection with ``using for``.
+
+## Important Bugfixes
+
+### Invalid Access for Calldata Parameters in Internal Library Functions
+
+**Problem**: The previous release, [Solidity 0.6.9](https://github.com/ethereum/solidity/releases/tag/v0.6.9), introduced the possibility to use ``calldata`` types in internal and not, as previously, only in external functions.
+This resulted in a bug related to calling library functions with calldata parameters via ``using for``.
+More specifically, the compiler would copy all calldata arguments to memory first - like done for calls to public library functions.
+The internal library function would then receive the memory pointer but interpret it as a calldata pointer resulting in
+calldata reads at the wrong location and potential stack corruption. The stack corruption is caused by the fact that
+calldata pointers can use two stack slots while memory pointers always only use one stack slot. This can in turn cause
+problems for other variables and a wrong jump at the function return.
+
+We assigned a severity of "very low" due to the fact that the bug was only introduced a week ago
+and is very easy to detect through testing.
+
+**Solution**: Upgrade to [Solidity 0.6.10](https://github.com/ethereum/solidity/releases/tag/v0.6.10), which does not perform the copy to memory.
+
+**Example of affected code**
+
+```solidity
+library L {
+    function helper(bytes calldata _x) internal pure returns (byte) {
+        return _x[0];
+    }
+}
+
+contract C {
+    using L for bytes;
+    function fun(bytes calldata _x) public pure returns (byte) {
+        // This would incorrectly copy ``_x`` from calldata
+        // to memory and then call L.helper with this memory array,
+        // but the function expects a calldata array and thus
+        // reads the data from the wrong calldata offset.
+        return _x.helper();
+    }
+}
+```
+We thank Luis from Gelato ([@gitpusha](https://github.com/gitpusha)) for reporting the bug!
+
+## Notable New Features 
+
+### Error Codes
+
+A notable new feature is the introduction of error codes. In Standard JSON input/output mode
+(``--standard-json``) the error codes will always be available under the property ``"errorCode"``.
+In the command-line interface, you can enable error codes via the switch ``--error-codes``.
+
+Similar errors (for example if error messages differ in type names only) might share the same error code. Vice versa, the same error message reported under different circumstances might have different error codes (although that is not a typical case).
+
+These error codes are intended to be reasonably stable across releases. However, there might be some exceptions, e.g. if one error is removed and another one added later, with the same error code.
```

Comment:
I don't really understand the example part @a3d4, it sounds a bitt off here `and another one added later, with the same error code.`?